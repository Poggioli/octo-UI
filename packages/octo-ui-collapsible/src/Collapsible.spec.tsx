import { render } from "@testing-library/react";
import {
  Collapsible,
  CollapsibleContent,
  CollapsibleTrigger
} from "./Collapsible";

describe("<Collapsible />", () => {
  it(`GIVEN a <Collapsible />
      WHEN render is called
      THEN should render a <Collapsible />`, () => {
    const rendered = render(
      <Collapsible>
        <CollapsibleTrigger>
          <button>Trigger</button>
        </CollapsibleTrigger>
        <CollapsibleContent>
          <p>Content p</p>
          <span>Content span</span>
        </CollapsibleContent>
      </Collapsible>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Collapsible />
      WHEN render is called
      THEN should render a <Collapsible />`, () => {
    const rendered = render(
      <Collapsible defaultOpen>
        <CollapsibleTrigger>
          <button>Trigger</button>
        </CollapsibleTrigger>
        <CollapsibleContent>
          <p>Content p</p>
          <span>Content span</span>
        </CollapsibleContent>
      </Collapsible>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Collapsible />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Collapsible.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <CollapsibleTrigger />
      WHEN toString is called
      THEN return a string`, () => {
    expect(CollapsibleTrigger.toString()).toBe(".octo-UI-c-PJLV");
  });

  it(`GIVEN a <CollapsibleContent />
      WHEN toString is called
      THEN return a string`, () => {
    expect(CollapsibleContent.toString()).toBe(".octo-UI-c-buTLZQ");
  });
});
