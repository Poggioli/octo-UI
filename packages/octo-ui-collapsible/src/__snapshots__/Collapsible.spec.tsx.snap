// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collapsible /> GIVEN a <Collapsible />
      WHEN render is called
      THEN should render a <Collapsible /> 1`] = `
<div>
  <div
    class="octo-UI-c-PJLV"
    data-state="closed"
  >
    <button
      aria-controls="radix-:r0:"
      aria-expanded="false"
      class="octo-UI-c-PJLV"
      data-state="closed"
      type="button"
    >
      Trigger
    </button>
    <div
      class="octo-UI-c-buTLZQ"
      data-state="closed"
      hidden=""
      id="radix-:r0:"
      style=""
    />
  </div>
</div>
`;

exports[`<Collapsible /> GIVEN a <Collapsible />
      WHEN render is called
      THEN should render a <Collapsible /> 2`] = `
<div>
  <div
    class="octo-UI-c-PJLV"
    data-state="open"
  >
    <button
      aria-controls="radix-:r1:"
      aria-expanded="true"
      class="octo-UI-c-PJLV"
      data-state="open"
      type="button"
    >
      Trigger
    </button>
    <div
      class="octo-UI-c-buTLZQ"
      data-state="open"
      id="radix-:r1:"
      style="transition-duration: 0s; animation-name: none;"
    >
      <p>
        Content p
      </p>
      <span>
        Content span
      </span>
    </div>
  </div>
</div>
`;
